CMakeLists.txt

```diff
@@ -1057,6 +1057,16 @@ else ()
   set (HDF5_ENABLE_MAP_API OFF CACHE BOOL "Build the map API" FORCE)
 endif ()
 
+#-----------------------------------------------------------------------------
+# Option to override the compiler for pkg-config & h5cc
+#-----------------------------------------------------------------------------
+if (HDF5_ENABLE_PARALLEL AND MPI_C_FOUND)
+  set (_PKG_CONFIG_C_COMPILER ${MPI_C_COMPILER})
+else ()
+  set (_PKG_CONFIG_C_COMPILER ${CMAKE_C_COMPILER})
+endif ()
+set (PKG_CONFIG_C_COMPILER ${_PKG_CONFIG_C_COMPILER} CACHE STRING "C compiler to use in h5cc")
```

Comment:
Bikeshedding a bit here, but as these are HDF5-specific options I'd generally prefer that they follow that naming. Maybe something like `HDF5_H5CC_C_COMPILER`, etc. `h5cc` using pkg-config is really just an implementation detail.

Reply:
Sure. I've just followed the original names used in the individual files but I definitely agree that better names are better :-).

Reply:
Done.

Reply:
Thanks! Also makes it a little clearer for me to read now compared to before:

`set (_PKG_CONFIG_C_COMPILER ...)` -> `set (PKG_CONFIG_C_COMPILER ${_PKG_CONFIG_C_COMPILER} ...)` -> `set (_PKG_CONFIG_COMPILER ${PKG_CONFIG_C_COMPILER})`